jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java

```diff
@@ -348,6 +387,140 @@ private static DiagnosticProperties doDump(String diagnosticCommand) {
 		return result;
 	}
 
+/*[IF JFR_SUPPORT]*/
+	private static Long convertToMilliseconds(String timeValue) {
+		long timeInMilli = 0L;
+
+		// extract the numeric part and the time unit
+		String numericPart = timeValue.replaceAll("[^0-9]", "");
+
+		// convert the unit to lowercase for consistency
+		String timeUnit = timeValue.replaceAll("[0-9]", "").toLowerCase();
+
+		// parse the numeric part
+		long time = Long.parseLong(numericPart);
+
+		// convert to milliseconds based on the unit
+		switch (timeUnit) {
+		case "s":
+			timeInMilli = TimeUnit.SECONDS.toMillis(time);
+			break;
+		case "m":
+			timeInMilli = TimeUnit.MINUTES.toMillis(time);
+			break;
+		case "h":
+			timeInMilli = TimeUnit.HOURS.toMillis(time);
+			break;
+		case "d":
+			timeInMilli = TimeUnit.DAYS.toMillis(time);
+			break;
+		default:
+			// no unit or unrecognized unit, assume milliseconds
+			timeInMilli = time;
+			break;
+		}
+		return Long.valueOf(timeInMilli);
```

Comment:
do we need to return `Long` why not `long`

Reply:
`Long != null` was used to determine if there is a `duration` sub-option, and replaced it with `long > 0` instead.